### PR TITLE
bluetooth: controller: Connection meta property support in LLL

### PR DESCRIPTION
**Pull request by mtpr-ot**
Added support for vendor specific meta data in LLL conn object. This
enables vendors to add state data to connection, for supporting
specialized BLE slave features.

Signed-off-by: Morten Priess <mtpr@oticon.com>

### CODE REVIEW

**Comment by mtpr-ot** [comment]
Feature config- and/or type naming may need to change to be in line with conventions (?)

File: subsys/bluetooth/controller/hci/hci.c [hunk 1]
```diff
@@ -28,6 +28,9 @@
 #include "hal/ccm.h"
 #include "ll_sw/pdu.h"
 #include "ll_sw/lll.h"
+#if defined(CONFIG_BT_CTLR_CONN_PROP_META)
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/hci/hci.c]: Having same guard here and inside the header file seems to be redundant.

File: subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c [hunk 2]
```diff
@@ -18,9 +18,16 @@
 #include "pdu.h"
 
 #include "lll.h"
+#if defined(CONFIG_BT_CTLR_CONN_PROP_META)
+#include "lll_conn_ext.h"
+#endif /* CONFIG_BT_CTLR_CONN_PROP_META */
 #include "lll_conn.h"
 #include "ull_conn_internal.h"
 
+#ifndef RADIO_TXP_DEFAULT
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c]: This seems to be an unrelated change. Move to a different PR?

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 3]
```diff
@@ -605,6 +606,9 @@ u8_t ll_adv_enable(u8_t enable)
 		conn_lll->slave.window_widening_event_us = 0;
 		conn_lll->slave.window_size_prepare_us = 0;
 		/* FIXME: END: Move to ULL? */
+#if defined(CONFIG_BT_CTLR_CONN_PROP_META)
+		memset(&conn_lll->prop_meta, 0, sizeof(conn_lll->prop_meta));
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c]: Maybe this should call a function to initialize the meta data instead of assuming that zero-initialization is the right thing to do.

Comment (mtpr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_adv.c]: > Maybe this should call a function to initialize the meta data instead of assuming that zero-initialization is the right thing to do.

I would prefer to avoid having vendor specific meta property functions as well. You can always structure your meta data such that setting all to '0' will be an unambiguous operation.

File: subsys/bluetooth/controller/hci/hci.c (hunk 1 above)
Comment (mtpr-ot) [reply subsys/bluetooth/controller/hci/hci.c]: > Having same guard here and inside the header file seems to be redundant.

OK, removing conditional #include

File: subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c (hunk 2 above)
Comment (mtpr-ot) [reply subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c]: > This seems to be an unrelated change. Move to a different PR?

Moved to a separate commit. Necessary for compiling the feature.

**Review by wopu-ot — APPROVED** [review]
LGTM

**Comment by carlescufi** [comment]
@cvinayak could you review please?

File: subsys/bluetooth/controller/Kconfig [hunk 4]
```diff
@@ -535,6 +535,13 @@ config BT_CTLR_LOW_LAT_ULL
 	  Low latency ULL implementation that uses tailchaining instead of while
 	  loop to demux rx messages from LLL.
 
+config BT_CTLR_CONN_PROP_META
```
Comment (cvinayak) [inline subsys/bluetooth/controller/Kconfig]: BT_CTLR_CONN_META, drop proprietary.

File: subsys/bluetooth/controller/hci/hci.c [hunk 5]
```diff
@@ -28,6 +28,9 @@
 #include "hal/ccm.h"
 #include "ll_sw/pdu.h"
 #include "ll_sw/lll.h"
+#if defined(CONFIG_BT_LL_SW_SPLIT)
+#include "lll_conn_ext.h"
```
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c]: include inside lll_conn.h, as it depends on lll_conn_ext.h

File: subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c [hunk 6]
```diff
@@ -18,9 +18,16 @@
 #include "pdu.h"
 
 #include "lll.h"
+#if defined(CONFIG_BT_LL_SW_SPLIT)
+#include "lll_conn_ext.h"
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c]: Same here, love lll_conn_ext.h into lll_conn.h

File: subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c]: Move this to top level subsys/bluetooth/controller/hal/radio.h

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn_ext.h [hunk 7]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2018-2019 Nordic Semiconductor ASA
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn_ext.h]: Fix copyright

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by cvinayak — APPROVED** [review]
LGTM

**Review by wopu-ot — APPROVED** [review]
